avcodec.go

```diff
@@ -58,6 +63,51 @@ func (d *avCodecDecoder) Description() string {
 	return fmt
 }
 
+func (d *avCodecDecoder) IsStreamable() bool {
+	var atomSize uint32
+	var atomType [4]byte
+	bytesRead := int64(0)
+
+	reader := bytes.NewReader(d.buf)
+
+	// Read atoms within the probe limit
+	for bytesRead < probeBytesLimit && bytesRead < int64(len(d.buf)) {
+		// Read atom header
+		err := binary.Read(reader, binary.BigEndian, &atomSize)
+		if err != nil {
+			break
+		}
+		err = binary.Read(reader, binary.BigEndian, &atomType)
+		if err != nil {
+			break
+		}
+		bytesRead += int64(atomHeaderSize)
+
+		// Check for 'moov' atom
+		if string(atomType[:]) == "moov" {
```

Comment:
does go do anything weird when making a string from invalid utf-8? might be safer to compare to a 4 byte array `'m', 'o', 'o', 'v'`

Reply:
Oh, great point. I'll ensure the C++ performs a byte-comparison of the atom name to avoid any string-interpretation.